core/embed/trezorhal/flash.h

```diff
@@ -26,11 +26,7 @@
 #include "flash_ll.h"
 #include "secbool.h"
 
-#ifndef TREZOR_EMULATOR
-#include STM32_HAL_H
-#endif
-
-#ifdef STM32U5
+#ifdef USE_FLASH_BURST
```

Comment:
My thought was to leave this here unconditional, and use USE_FLASH_BURST in flash_area where we use #ifdef FLASH_BURST_SIZE. Or we can leave the condition here, but in any case i think it's preferable to replace the condition in flash_area with USE_FLASH_BURST too.